CMakeLists.txt

```diff
@@ -189,6 +189,31 @@ set(OPENCV_LIBS
     "libjpeg.so"
 )
 
+# In CPU-only mode, get the hpcx libraries neededby libtorch.so.
+if(NOT TRITON_ENABLE_GPU)
+  set(HPCX_LIBS
+      "libucc.so.1"
+      "libucm.so.0"
+      "libucp.so.0"
+      "libucs.so.0"
+      "libuct.so.0"
+  )
+  add_custom_command(
+    OUTPUT
+      ${HPCX_LIBS}
+    COMMAND docker pull ${TRITON_PYTORCH_DOCKER_IMAGE}
+    COMMAND docker rm pytorch_backend_hpcxlib || echo "error ignored..." || true
+    COMMAND docker create --name pytorch_backend_hpcxlib ${TRITON_PYTORCH_DOCKER_IMAGE}
+    COMMAND docker cp -L pytorch_backend_hpcxlib:/opt/hpcx/ucc/lib/libucc.so.1 libucc.so.1
+    COMMAND docker cp -L pytorch_backend_hpcxlib:/opt/hpcx/ucx/lib/libucm.so.0 libucm.so.0
+    COMMAND docker cp -L pytorch_backend_hpcxlib:/opt/hpcx/ucx/lib/libucp.so.0 libucp.so.0
+    COMMAND docker cp -L pytorch_backend_hpcxlib:/opt/hpcx/ucx/lib/libucs.so.0 libucs.so.0
+    COMMAND docker cp -L pytorch_backend_hpcxlib:/opt/hpcx/ucx/lib/libuct.so.0 libuct.so.0
+    COMMAND docker rm pytorch_backend_hpcxlib
+    VERBATIM
```

Comment:
Instead of modifying cmake can we copy over the whole `/opt/hpcx/ucc/` and `/opt/hpcx/ucx` in the build.py?

Reply:
Sure, I will try this approach.